doc/user/content/ops/deployment.md

```diff
@@ -12,61 +12,137 @@ issue](https://github.com/MaterializeInc/materialize/issues/new?labels=C-feature
 
 ## Memory
 
-Materialize stores the majority of its state in-memory, and works best when the streamed data
-can be reduced in some way. For example, if you know that only a subset of your rows and columns
-are relevant for your queries, it helps to avoid materializing sources or views until you've
-expressed this to the system (we can avoid stashing that data, which can in some cases dramatically
-reduce the memory footprint).
+Materialize stores the majority of its state in memory, and works best when the
+streamed data can be reduced in some way. For example, if you know that only a
+subset of your rows and columns are relevant for your queries, it helps to avoid
+materializing sources or views until you've expressed this to the system.
+Materialize can then avoid stashing the full set of rows and columns, which can
+in some cases dramatically reduce Materialize's memory footprint.
 
-To minimize the chances that Materialize runs out of memory in a production environment,
-we recommend you make additional memory available to Materialize via a SSD-backed
-swap file or swap partition.
+### Compaction
 
-This is particularly important in Linux and in Docker, where swap may not be automatically
-setup for you:
+To prevent memory from growing without bound, Materialize periodically
+"compacts" data in [arrangements](/overview/api-components#indexes). For
+example, if you have a source that tracks product inventory, you might receive
+periodic inventory updates throughout the day:
+
+```
+(T-shirts, 9:07am, +500)
+(T-shirts, 11:32am, -1)
+(T-shirts, 3:14pm, -2)
+```
+
+Logical compaction will fold historical updates that fall outside the compaction
+window into the state at the start of the window.
+
+```
+(T-shirts, 3:14pm, +497)
+```
+
+Materialize will only perform this compaction on data that falls outside the
+logical compaction window. The default compaction window is 60 seconds, but the
+window can be adjusted via the
+[`--logical-compaction-window`](/cli/#compaction-window) option.
```

Comment:
Maybe minor, but perhaps we should be explicit that the "window" is trailing the most recently available time. At least, I think some folks would expect window to type check as having bounds, rather than being a time. Perhaps just "The default compaction window is 60 seconds behind the current time".

Reply:
Makes total sense! Done.